tests/Functional/Driver/PDO/PgSQL/DBAL6024Test.php

```diff
@@ -0,0 +1,35 @@
+<?php
+
+namespace Doctrine\DBAL\Tests\Functional\Driver\PDO\PgSQL;
+
+use Doctrine\DBAL\Schema\Comparator;
+use Doctrine\DBAL\Schema\Table;
+use Doctrine\DBAL\Tests\FunctionalTestCase;
+
+class DBAL6024Test extends FunctionalTestCase
+{
+    public function testDropPrimaryKey(): void
+    {
+        $table = new Table('mytable');
+        $table->addColumn('id', 'integer');
+        $table->setPrimaryKey(['id']);
+        $this->dropAndCreateTable($table);
+
+        $newTable = clone $table;
+        $newTable->dropPrimaryKey();
+
+        $diff = (new Comparator())->compareTables($table, $newTable);
+
+        $statements = $this->connection->getDatabasePlatform()->getAlterTableSQL($diff);
+        foreach ($statements as $statement) {
+            $this->connection->executeStatement($statement);
+        }
+
+        $schemaManager = $this->connection->createSchemaManager();
```

Comment:
Sorry, I missed that you're calling `new Comparator()` directly. Let's use `createComparator()` instead.

```suggestion
        $schemaManager = $this->connection->createSchemaManager();
        $diff          = $schemaManager->createComparator()->compareTables($table, $newTable);

        $statements = $this->connection->getDatabasePlatform()->getAlterTableSQL($diff);
        foreach ($statements as $statement) {
            $this->connection->executeStatement($statement);
        }
```